core/templates/domain/summary/exploration-summary-backend-api.service.ts

```diff
@@ -17,77 +17,107 @@
+  'summaries': [{
+    'category': string,
+    'community_owned': boolean,
+    'id': string,
+    'language_code': string,
+    'num_views': number,
+    'objective': string,
+    'status': string,
+    'tags': [],
+    'thumbnail_bg_color': string,
+    'thumbnail_icon_url': string,
+    'title': string
+  }]
+}
+@Injectable({
+  providedIn: 'root'
+})
+export class ExplorationSummaryBackendApiService {
+  constructor(
+    private httpClient: HttpClient,
+    private alertsService: AlertsService,
+    private validatorsService: ValidatorsService) {}
 
-        if (errorCallback) {
-          errorCallback(returnValue);
-        }
-        return;
+  private _fetchExpSummaries(
+      explorationIds: string[],
+      includePrivateExplorations: boolean,
+      successCallback: (
+        value?: ExplorationSummaryBackendDict |
+        PromiseLike<ExplorationSummaryBackendDict>) => void,
+      errorCallback: (reason?: string[]) => void): void {
+    if (!explorationIds.every(expId =>
+      this.validatorsService.isValidExplorationId(expId, true))) {
+      this.alertsService.addWarning('Please enter a valid exploration ID.');
+      let returnValue = [];
+      for (let i = 0; i < explorationIds.length; i++) {
+        returnValue.push(null);
       }
-
-      var explorationSummaryDataUrl = EXPLORATION_SUMMARY_DATA_URL_TEMPLATE;
-
-      $http.get(explorationSummaryDataUrl, {
-        params: {
-          stringified_exp_ids: JSON.stringify(explorationIds),
-          include_private_explorations: JSON.stringify(
-            includePrivateExplorations)
-        }
-      }).then(function(response) {
-        var summaries = angular.copy(response.data.summaries);
+      if (errorCallback) {
+        errorCallback(returnValue);
+      }
+      return;
+    }
+    const explorationSummaryDataUrl =
+    AppConstants.EXPLORATION_SUMMARY_DATA_URL_TEMPLATE;
+    this.httpClient.get(explorationSummaryDataUrl, {
+      params: {
+        stringified_exp_ids: JSON.stringify(explorationIds),
+        include_private_explorations: JSON.stringify(
+          includePrivateExplorations)
+      }
+    }).toPromise().then((summaries: ExplorationSummaryBackendDict) => {
+      try {
         if (successCallback) {
           if (summaries === null) {
-            var summariesError = (
+            const summariesError = (
               'Summaries fetched are null for explorationIds: ' + explorationIds
             );
             throw new Error(summariesError);
+          } else if (typeof (summaries) === 'string') {
+            const fetchingError = (
+              'Error on loading public exploration summaries.');
+            throw new Error(fetchingError);
           }
           successCallback(summaries);
         }
-      })['catch'](function(errorResponse) {
+      } catch (errorResponse) {
         if (errorCallback) {
-          errorCallback(errorResponse.data || errorResponse);
+          errorCallback(errorResponse);
         }
```

Comment:
@kevintab95 any reason for this convoluted logic? We ourselves are throwing errors and catching them? Why can't we just use errorCallBack() followed by a return statement?

Reply:
Hm, right. Yep, I think we can get rid of the try catch and use the errorCallback where required instead. Thanks!

Reply:
Removed!